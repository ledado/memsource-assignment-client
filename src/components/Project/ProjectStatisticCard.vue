<template>
  <v-row no-gutters class="project-statistic-card" :class="backgroundClass">
    <v-col cols="12">
      <v-row no-gutters>
        <v-col cols="12" class="project-statistic-card-text body-2">
          {{ title }}
        </v-col>
        <v-col cols="12" class="project-statistic-card-text text-h4">
          {{ value }}
        </v-col>
        <v-col cols="12" class="project-statistic-card-text body-2">
          {{ description }}
        </v-col>
      </v-row>
    </v-col>
  </v-row>
</template>

<script lang="ts">
import {Component, Prop, Vue} from 'vue-property-decorator';

@Component
export default class ProjectStatisticCard extends Vue {
  @Prop() value!: number | string;
  @Prop() title!: string;
  @Prop() description!: string;
  @Prop() backgroundClass!: string;
}
</script>
<style lang="scss" scoped>
.project-statistic-card {
  border: 1px solid black;
  border-radius: 4px;
  padding: 18px 16px;
  &.orange {
    background: #bf360c;
  }
  &.green {
    background: #2e7d32;
  }
  &.blue {
    background: #0277bd;
  }

  .project-statistic-card-text {
    color: white;
  }
}
</style>
