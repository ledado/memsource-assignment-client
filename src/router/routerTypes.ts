export const routerTypes = {
  MAIN: 'main',
  PROJECT_NEW: 'project-new',
  PROJECT_UPDATE: 'project-update',
};
