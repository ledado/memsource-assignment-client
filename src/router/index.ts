import Vue from 'vue';
import VueRouter, {RouteConfig} from 'vue-router';
import {errorBus} from '@/service/events/ErrorBus';
import {layoutTypes} from '@/layouts/layoutTypes';
import {routerTypes} from '@/router/routerTypes';

Vue.use(VueRouter);

const routes: Array<RouteConfig> = [
  {
    path: '/',
    name: routerTypes.MAIN,
    component: () => import('../views/Main.vue'),
    meta: {
      layout: layoutTypes.DEFAULT,
    },
  },
  {
    path: '/project/new',
    name: routerTypes.PROJECT_NEW,
    component: () => import('../views/Project/MainProjectCreate.vue'),
    meta: {
      layout: layoutTypes.DEFAULT,
    },
  },
  {
    path: '/project/:id',
    name: routerTypes.PROJECT_UPDATE,
    component: () => import('../views/Project/MainProjectUpdate.vue'),
    meta: {
      layout: layoutTypes.DEFAULT,
    },
    props: true,
  },
  {
    path: '*',
    component: () => import('../views/Error/Error404.vue'),
  },
];

const router = new VueRouter({
  mode: 'history',
  base: process.env.BASE_URL,
  routes,
});

// error state clean up
router.afterEach((to, from) => {
  if (from.name && errorBus.state) {
    errorBus.errorCode = undefined;
    errorBus.state = false;
  }
});

export default router;
