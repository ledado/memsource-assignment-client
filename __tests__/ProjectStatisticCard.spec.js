import {shallowMount} from '@vue/test-utils';
import ProjectStatisticCard from '@/components/Project/ProjectStatisticCard.vue';
import {routerTypes} from '@/router/routerTypes';

describe('ProjectStatisticCard.vue', () => {
  it('render component', () => {
    const wrapper = shallowMount(ProjectStatisticCard, {
      propsData: {
        value: 10,
        title: 'Test',
        description: 'Description test',
        backgroundClass: 'orange',
      },
      stubs: ['router-link'],
      mocks: {
        $t: (msg) => msg,
      },
      mixins: [
        {
          data: () => {
            return {
              routerTypes,
            };
          },
        },
      ],
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
