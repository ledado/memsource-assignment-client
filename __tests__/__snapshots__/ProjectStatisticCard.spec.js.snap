// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectStatisticCard.vue render component 1`] = `
"<v-row-stub tag=\\"div\\" nogutters=\\"true\\" class=\\"project-statistic-card orange\\">
  <v-col-stub cols=\\"12\\" tag=\\"div\\">
    <v-row-stub tag=\\"div\\" nogutters=\\"true\\">
      <v-col-stub cols=\\"12\\" tag=\\"div\\" class=\\"project-statistic-card-text body-2\\">
        Test
      </v-col-stub>
      <v-col-stub cols=\\"12\\" tag=\\"div\\" class=\\"project-statistic-card-text text-h4\\">
        10
      </v-col-stub>
      <v-col-stub cols=\\"12\\" tag=\\"div\\" class=\\"project-statistic-card-text body-2\\">
        Description test
      </v-col-stub>
    </v-row-stub>
  </v-col-stub>
</v-row-stub>"
`;
